48 combine built in parsers into one document

stephffuller opened the pull request:
@mwl I need some help resolving the aliases in this. All of the separate pages for built-in parsers were combined into one page here. I tried to pull the same types of aliases that were used in other files so that other references to the old separate pages would work, but something isn't right. Any guidance would be helpful.

mwl commented on content/developer-resources/integrations/other/nginx.md:
```diff
@@ -44,7 +44,7 @@ output.elasticsearch:
 See the page on [Filebeat]({{< ref "filebeat.md" >}}) for further details.
 
-The above Filebeat configuration uses the [built-in parser `accesslog`]({{< ref "accesslog.md" >}}).
+The above Filebeat configuration uses the [built-in parser `accesslog`]({{< ref "parsers/built-in-parsers/_index.md/#accesslog" >}}).
```
Same again. Looks like I'm ending up on the same page.

mwl commented on content/operations-guide/parsers/built-in-parsers/builtin-parsers.md:
```diff
@@ -0,0 +1,169 @@
+title: Built-in-Parsers
+aliases:
+- /accesslog.md/
+- /bro-json.md/
```
```suggestion
- /operations-guide/parsers/built-in-parsers/bro-json
- /parsers/built-in-parsers/bro-json
```

mwl commented on content/operations-guide/parsers/built-in-parsers/builtin-parsers.md:
```diff
@@ -0,0 +1,169 @@
+aliases:
+- /accesslog.md/
+- /bro-json.md/
+- /corelight-json.md/
```
```suggestion
- /operations-guide/parsers/built-in-parsers/corelight-json
- /parsers/built-in-parsers/corelight-json
```

mwl commented on content/operations-guide/parsers/built-in-parsers/builtin-parsers.md:
```diff
@@ -0,0 +1,169 @@
+- /accesslog.md/
+- /bro-json.md/
+- /corelight-json.md/
+- /json.md/
```
```suggestion
- /operations-guide/parsers/built-in-parsers/json
- /parsers/built-in-parsers/json
```

mwl commented on content/operations-guide/parsers/built-in-parsers/builtin-parsers.md:
```diff
@@ -0,0 +1,169 @@
+- /bro-json.md/
+- /corelight-json.md/
+- /json.md/
+- /kv-millis.md/
```
```suggestion
- /operations-guide/parsers/built-in-parsers/kv-millis
- /parsers/built-in-parsers/kv-millis
```

mwl commented on content/operations-guide/parsers/built-in-parsers/builtin-parsers.md:
```diff
@@ -0,0 +1,169 @@
+- /corelight-json.md/
+- /json.md/
+- /kv-millis.md/
+- /kv.md/
```
```suggestion
- /operations-guide/parsers/built-in-parsers/kv
- /parsers/built-in-parsers/kv
```

mwl commented on content/operations-guide/parsers/built-in-parsers/builtin-parsers.md:
```diff
@@ -0,0 +1,169 @@
+- /kv.md/
+- /syslog.md/
+- /syslog-utc.md/
+- /serilog.md/
```
```suggestion
- /operations-guide/parsers/built-in-parsers/serilog
- /parsers/built-in-parsers/serilog
```

mwl commented on content/operations-guide/parsers/built-in-parsers/builtin-parsers.md:
```diff
@@ -0,0 +1,169 @@
+- /kv-millis.md/
+- /kv.md/
+- /syslog.md/
+- /syslog-utc.md/
```
```suggestion
- /operations-guide/parsers/built-in-parsers/syslog-utc
- /parsers/built-in-parsers/syslog-utc
```

mwl commented on content/operations-guide/parsers/built-in-parsers/builtin-parsers.md:
```diff
@@ -0,0 +1,169 @@
+- /json.md/
+- /kv-millis.md/
+- /kv.md/
+- /syslog.md/
```
```suggestion
- /operations-guide/parsers/built-in-parsers/syslog
- /parsers/built-in-parsers/syslog
```

mwl commented on content/operations-guide/use_cases/getting-started-application-logs.md:
```diff
@@ -23,8 +23,7 @@ Filebeat sends logs as unstructured text. To parse these logs once they arrive a
 
-For now let os assume we can use the build-in [`kv`]({{< ref "kv.md" >}}) (Key-Value) parser, it extracts fields
-from the log lines, anything of the form `key=value`.
+For now let os assume we can use the build-in [`kv`]({{< relref "parsers/built-in-parsers/_index.md#kv" >}}) (Key-Value) parser, it extracts fields from the log lines, anything of the form `key=value`.
```
Broken link

mwl commented on content/tutorial/_index.md:
```diff
@@ -93,16 +93,13 @@ _Tip: If you are already using ElasticSearch ELK you can also take a look at how
-and can rip out almost anything with the [`kv`]({{< relref "../operations-guide/parsers/built-in-parsers/kv.md" >}}). parser, you 
-may have special needs for your custom application logs. Fortunately Humio allows you to 
-create your own custom parsers as documented here: [create your own custom parser]({{< relref "../operations-guide/parsers/creating-a-parser.md" >}}).
+While Humio has built in support for the most popular logging formats (e.g. AccessLog, JSON), and can rip out almost anything with the [`kv`]({{< relref "../operations-guide/parsers/built-in-parsers/_index.md/#kv" >}}) parser, you may have special needs for your custom application logs. Fortunately, Humio allows you to create your own [custom parsers]({{< relref "../operations-guide/parsers/creat ...
```
The ``[`kv`]({{< relref "../operations-guide/parsers/built-in-parsers/_index.md/#kv" >}})`` link is broken

mwl commented:
@stephffuller As for the broken links, there's definitely some confusion there. I have to pages in my tree
- content/operations-guide/parsers/built-in-parsers/_index.md
- content/operations-guide/parsers/built-in-parsers/builtin-parsers.md

I guess the first one is legacy?
If so, delete it, and move `content/operations-guide/parsers/built-in-parsers/builtin-parsers.md` up, to `content/operations-guide/parsers/builtin-parsers.md`.
Then you can refer to the page as ``[`kv`]({{< ref "built-in-parsers.md#kv" >}})``

Also, is it `built-in` or `builtin`? :)

mwl commented:
@stephffuller Made a few edits to the [comment above](https://github.com/humio/docs.humio.com/pull/152#issuecomment-528236667). So make sure you read it from GitHub.

stephffuller commented:
@mwl I incorporated your changes. I think. Please make sure you update your local branch. I actually deleted the built-in-parsers.md file and merged it and _index.md yesterday, so that should be the file you're looking at. If you could review quickly one last time. I also believe all the links you were saying were broken are resolved. They link to different sections in the same document, instead of separate files. Thanks!

hobbes-the-tiger commented:
> @stephffuller As for the broken links, there's definitely some confusion there. I have two pages in my tree
> 
>     * `content/operations-guide/parsers/built-in-parsers/_index.md`
> 
>     * `content/operations-guide/parsers/built-in-parsers/builtin-parsers.md`
> 
> 
> I guess the first one is legacy?
> If so, delete it, and move `content/operations-guide/parsers/built-in-parsers/builtin-parsers.md` up, to become `content/operations-guide/parsers/built-in-parsers.md`.
> Then you can refer to the page as `` [`kv`]({{< ref "built-in-parsers.md#kv" >}}) ``
> 
> Also, is it `built-in` or `builtin`? :)

I think @LH-ExcellentAdventure can let us know if built-in or builtin is what we use.

stephffuller commented:
No, the second file has been removed.
content/operations-guide/parsers/built-in-parsers/_index.md is the only
current file that exists now.

-Steph

On Thu, Sep 5, 2019 at 4:20 PM Anne Henmi <notifications@github.com> wrote:

> @stephffuller <https://github.com/stephffuller> As for the broken links,
> there's definitely some confusion there. I have two pages in my tree
>
> * `content/operations-guide/parsers/built-in-parsers/_index.md`
>
> * `content/operations-guide/parsers/built-in-parsers/builtin-parsers.md`
>
> I guess the first one is legacy?
> If so, delete it, and move
> content/operations-guide/parsers/built-in-parsers/builtin-parsers.md up,
> to become content/operations-guide/parsers/built-in-parsers.md.
> Then you can refer to the page as [`kv`]({{< ref "built-in-parsers.md#kv"
> >}})
>
> Also, is it built-in or builtin? :)
>
> I think @LH-ExcellentAdventure <https://github.com/LH-ExcellentAdventure>
> can let us know if built-in or builtin is what we use.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/humio/docs.humio.com/pull/152?email_source=notifications&email_token=AMO6GFFQ2BYYIVT44DXMMDTQIFSY3A5CNFSM4ISFXNSKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD6AUEJY#issuecomment-528564775>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AMO6GFD7NBIQFBS6ALC3NDLQIFSY3ANCNFSM4ISFXNSA>
> .
>


-- 

Stephanie Fuller <stephanie@humio.com>
Sr. Technical Content Developer, Product Education
Humio <https://www.humio.com/>

LH-ExcellentAdventure commented:
> Also, is it `built-in` or `builtin`? :)
Per merriam-webster dictionary, it's "built-in". I'll get that in the glossary. =)

hobbes-the-tiger commented:
Not sure what happened, but /operations-guide/parsers/_index.md graphic is not rendering correctly in this branch. That needs to be fixed, along with @mwl's comments before moving forward.

stephffuller commented:
@mwl  I think I know what happened. I was working late Wednesday (my evening), which must have been when you started working on it yourself (your Thurs morning).

stephffuller commented:
@mwl @hobbes-the-tiger Ok, I'm done making changes for now. I *think* I have addressed all the issues. Annie, I was able to see the graphic on my local "make run". Not sure what the issue there was?

mwl commented on content/tutorial/_index.md:
```diff
@@ -93,16 +93,13 @@ _Tip: If you are already using ElasticSearch ELK you can also take a look at how
-and can rip out almost anything with the [`kv`]({{< relref "../operations-guide/parsers/built-in-parsers/kv.md" >}}). parser, you 
-may have special needs for your custom application logs. Fortunately Humio allows you to 
-create your own custom parsers as documented here: [create your own custom parser]({{< relref "../operations-guide/parsers/creating-a-parser.md" >}}).
+While Humio has built in support for the most popular logging formats (e.g. AccessLog, JSON), and can rip out almost anything with the [kv]({{< relref "../operations-guide/parsers/built-in-parsers/#kv" >}}) parser, you may have special needs for your custom application logs. Fortunately, Humio allows you to create your own [custom parsers]({{< relref "../operations-guide/parsers/creating-a-parser ...
```
```suggestion
While Humio has built in support for the most popular logging formats (e.g. AccessLog, JSON), and can rip out almost anything with the [kv]({{< ref "built-in-parsers.md#kv" >}}) parser, you may have special needs for your custom application logs. Fortunately, Humio allows you to create your own [custom parsers]({{< ref "creating-a-parser.md" >}}).
```

mwl commented on content/tutorial/_index.md:
```diff
@@ -93,16 +93,13 @@ _Tip: If you are already using ElasticSearch ELK you can also take a look at how
 - [Language Syntax]({{< ref "language-syntax/_index.md" >}})
 - [Query Function Reference]({{< ref "query-functions/_index.md" >}})
-- [Creating Custom Parsers]({{< ref "../operations-guide/parsers/_index.md" >}})
+- [Creating Custom Parsers]({{< relref "../operations-guide/parsers/creating-a-parser.md" >}})
```
```suggestion
- [Creating Custom Parsers]({{< ref "creating-a-parser.md" >}})
```

stephffuller commented on content/operations-guide/parsers/built-in-parsers/_index.md:
```diff
@@ -2,23 +2,173 @@
 title: "Built-in Parsers"
```
Done.

mwl replied:
It doesn't look like the page wasn't deleted?

stephffuller commented:
Ok. I moved the content to built-in-parsers.md as suggested by @mwl . I believe it's now ready for @LH-ExcellentAdventure to work her magic.

stephffuller commented:
@LH-ExcellentAdventure This was pushed to staging

mwl commented on content/operations-guide/parsers/built-in-parsers/built-in-parsers.md:
```diff
@@ -0,0 +1,168 @@
+---
```
Please move the page to `content/operations-guide/parsers/built-in-parsers.md`

stephffuller replied:
Done. And removed the built-in-parsers subfolder.

stephffuller commented:
@mwl  Please let me know if there is anything outstanding with your review.

mwl commented on content/tutorial/_index.md:
```diff
@@ -93,16 +93,13 @@ _Tip: If you are already using ElasticSearch ELK you can also take a look at how
 - [Language Syntax]({{< relref "developer-resources/language-syntax/_index.md" >}})
 - [Query Function Reference]({{< relref "developer-resources/query-functions/_index.md" >}})
-- [Creating Custom Parsers]({{< relref "operations-guide/parsers/_index.md" >}})
+- [Creating Custom Parsers]({{< relref "operations-guide/parsers/creating-a-parser.md" >}})
```
```suggestion
- [Creating Custom Parsers]({{< ref "creating-a-parser.md" >}})
```

stephffuller replied:
@mwl Yeah, those changes were related to another issue 105 I was working on. If I had made them in the other branch, I would have had conflicts when I went to merge, as the changes were to a parser file that was deleted in this branch. So I went ahead and made them here in the combined document.

stephffuller commented:
@LH-ExcellentAdventure It should be good to go now. You can merge to master.
